features/FxA-54-push-on-status-change/README.md

```diff
@@ -0,0 +1,84 @@
+# Push Notifications for Account Status Changes
+
+https://mozilla.aha.io/features/FXA-54
+
+## Stories
+
+As a user signed in to an FxA connected device,
+I want my device to update its UI in a timely manner
+when I interact with my account from other devices.
+
+As a developer of client code for a connected device,
+I want to easily receive notifications so that I can
+respond to account status changes without resorting to
+polling the server.
+
+As a developer of Firefox Accounts,
+I want to stop Firefox from polling for account verification
+because this is the single biggest source
+of system load in production.
+
+## Success Criteria
+
+This feature will be successful if
+we successfully eliminate polling for account verification
+from the browser code on Desktop and Fennec.
+
+As a concrete metric
+we can measure the time between account verification
+and the use of a keyFetchToken by the browser,
+an action that it will only perform
+when it knows the account has been verified.
+
+## Details
+
+When FxA-45 lands,
+connected devices will be able to register a push API endpoint
+to receive notifications from the FxA server.
+We will make use of this facility
+to notify the device of account status changes.
+
+The WebPush specification allows messages to be sent
+with or without a payload.
+To keep things simple
+we will use an empty message to indicate
+that there has been a change
+in the core account status information,
+without communicating any more precise details
+of what that change may have been.
+This will give us an opportunity to gain experience
+with using push notifications in production,
+without having to commit to and build
+all the details of a full notification/communication protocol.
+
+A future feature will define the semantics
+of push notifications with a non-empty payload.
+
+When a connected device receives
+a push notification on its registered endpoint,
+and the notification has an empty payload,
+the device should perform the following checks
+to update its view of the account status:
+
+*  Check account status to see if it's verified, or deleted, or email has been changed
+*  If that fails with a 401, its session token has been revoked somehow.
+*  Check its device registration record for updated name
+*  Check for new or removed peer devices in the devices list
+*  Refresh profile information?
+
+
+The fxa-auth-server will send an empty-payload message
+to all registered devices
+in response to the following actions:
+
+* User verifying their account
+* User deleting their account
+* A new device connection being recorded
+* A device connection being deleted
+* Any action that invalidates a sessionToken
+* ...others?
+
+## Open Questions
+
+* How closely should we spec the client's update behaviour?
+* Do we need some fallback polling mechanism of the notification stuff gets in a bad state?
```

Comment:
`of` => `if`

I brought this question up and heard "if the push servers go down, we have bigger problems". If running the two branches in parallel for an Fx release or two is easy, it may be prudent while any push service scaling issues are ironed out. If running both branches is difficult or adds a significant amount of overhead, meh.